cloud_utils/scheduler/kubernetes_connector.py

```diff
@@ -85,14 +85,22 @@ def create_cron_job(
         ),
     )
     try:
-        options = {"namespace": "default", "body": cron_job, "pretty": "true"}
+        options = {"name": f"{pod_name}-cronjob", "namespace": "default", "body": cron_job, "pretty": "true"}
         if dry_run:
             options["dry_run"] = "All"
-        api_response = client.BatchV1beta1Api().create_namespaced_cron_job(**options)
-        logging.info(f"CronJob created: {api_response}")
+        api_response = client.BatchV1beta1Api().patch_namespaced_cron_job(**options)
+        logging.info(f"CronJob updated: {api_response}")
     except rest.ApiException as e:
-        logging.error(f"Error creating CronJob: {e}")
-        raise
+        logging.info(f"CronJob {options['name']} does'nt exist, creating...")
+        try:
+            options = {"namespace": "default", "body": cron_job, "pretty": "true"}
+            if dry_run:
+                options["dry_run"] = "All"
+            api_response = client.BatchV1beta1Api().create_namespaced_cron_job(**options)
+            logging.info(f"CronJob created: {api_response}")
+        except rest.ApiException as e:
+            logging.error(f"Error creating CronJob: {e}")
+            raise
```

Comment:
if you are just raising, I don't think there is a need for the try catch. we will see the error anyway so let's remove it.